org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/codeActions/ICodeActionService.xtend

```diff
@@ -15,12 +15,26 @@ import org.eclipse.lsp4j.jsonrpc.messages.Either
 import org.eclipse.xtext.ide.server.Document
 import org.eclipse.xtext.resource.XtextResource
 import org.eclipse.xtext.util.CancelIndicator
+import org.eclipse.xtext.ide.server.ILanguageServerAccess
 
 /**
  * @author Sven Efftinge - Initial contribution and API
+ * @deprecated use {@link ICodeActionService2} instead
  */
+@Deprecated
 interface ICodeActionService {
 	
 	def List<Either<Command, CodeAction>> getCodeActions(Document document, XtextResource resource, CodeActionParams params, CancelIndicator indicator)
+		
+}
+
+/**
+ * @author Jan Koehnlein
+ * @since 2.18
+ */
+ interface ICodeActionService2 {
 	
-}
\ No newline at end of file
+	def List<Either<Command, CodeAction>> getCodeActions(Document document, XtextResource resource, ILanguageServerAccess access, CodeActionParams params, CancelIndicator indicator)
```

Comment:
One question; since we introduce a new service and deprecate existing APIs, wouldn't it be better to have something like:

```java
def List<Either<Command, CodeAction>> getCodeActions(CodeActionOptions options)
```

where `CodeActionOptions` is
```java
@Data
class CodeActionOptions {
  val Document document;
  val XtextResource resource;
  val ILanguageServerAccess access;
  val CodeActionParams params;
  val CancelIndicator cancelIndicator;
}
```

otherwise, we might see `ICodeActionService3` popping up in the near future. Thoughts?

Reply:
should we do this for rename too?

Reply:
Already happened https://github.com/eclipse/xtext-core/pull/1091